src/plugins/region_map/public/components/custom_vector_upload.tsx

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, Fragment } from 'react';
+import axios from 'axios';
+import {
+  EuiButton,
+  EuiFilePicker,
+  EuiText,
+  EuiSpacer,
+  EuiCard,
+  EuiFieldText,
+  EuiSelect,
+  EuiTextColor,
+  EuiFormRow,
+} from '@elastic/eui';
+import { Services, getServices } from '../services';
+
+export type CustomVectorUploadProps = {
+  getServiceSettings: () => Promise<IServiceSettings>;
+} & VisOptionsProps<RegionMapVisParams>;
+
+function CustomVectorUpload(props: CustomVectorUploadProps) {
+  const services = getServices(props.vis.http);
+  const notifications = props.vis.notifications;
+  const [large, setLarge] = useState(true);
+
+  const INDEX_NAME_SUFFIX = '-map';
+  const INDEX_NAME_UPPERCASE_CHECK = /[A-Z]+/;
+  const INDEX_NAME_SPECIAL_CHARACTERS_CHECK = /[`!@#$%^&*()\=\[\]{};':"\\|,.<>\/?~]/;
+  const INDEX_NAME_NOT_BEGINS_WITH_CHECK = /^[-+_.]/;
+  const INDEX_NAME_BEGINS_WITH_CHECK = /^[a-z]/;
+  const INDEX_NAME_NOT_ENDS_WITH_CHECK = /.*-map$/;
+
+  // File size that can be uploaded: 250 MB
+  const MAX_FILE_SIZE_IN_BYTES = 26214400;
+  const MAX_LENGTH_OF_INDEX_NAME = 250;
+
+  const options = [
+    { value: 'geo_shape', text: 'Geo shape' },
+    { value: 'geo_point', text: 'Geo point' },
+  ];
+
+  const [value, setValue] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const onSelectChange = (e) => {
+    setSelectValue(e.target.value);
+  };
+
+  const [selectValue, setSelectValue] = useState('');
+
+  const onTextChange = (e) => {
+    setValue(e.target.value);
+  };
+
+  const onChange = (files) => {
+    if (files[0]) {
+      validateFileSize(files);
+    }
+  };
+
+  const fetchElementByName = (elementName) => {
+    return document.getElementsByName(elementName)[0];
+  };
+
+  const validateIndexName = (typedIndexName, isIndexNameWithSuffix) => {
+    let error = '';
+    const errorIndexNameDiv = fetchElementByName('errorIndexName');
+
+    // check for presence of index name entered by the user
+    if (!typedIndexName) {
+      error = 'Required';
+    } else {
+      // check for restriction on length of the index name
+      if (MAX_LENGTH_OF_INDEX_NAME < typedIndexName.length) {
+        error += ' Map name should be less than 250 characters.';
```

Comment:
How about add to new line for each error?

Reply:
yup will add that!

Reply:
shall we use constant in message as well instead of hard coding the number?